modules/core/src/copy.cpp

```diff
@@ -859,15 +861,16 @@ void repeat(InputArray _src, int ny, int nx, OutputArray _dst)
     CV_Assert( _src.dims() <= 2 );
     CV_Assert( ny > 0 && nx > 0 );
 
-    Size ssize = _src.size();
-    _dst.create(ssize.height*ny, ssize.width*nx, _src.type());
-
 #if !defined __APPLE__
-    CV_OCL_RUN(_dst.isUMat(),
+    CV_OCL_RUN(_src.isUMat(),
```

Comment:
`_dst.isUMat()` check is required for `CV_OCL_RUN` macro.

Also OpenCL(UMat) path doesn't support "inplace" mode.
Please take a look here: #8352 - it can't be resolved in general.